lib/debug/session.rb

```diff
@@ -98,6 +98,11 @@ def initialize ui
       @tp_thread_begin.enable
     end
 
+    def set_ui(ui)
+      @ui = ui
+      @management_threads << @ui.reader_thread if @ui.respond_to? :reader_thread
```

Comment:
Because:
- only remote ui needs an extra thread for reader
- the default ui is console ui

we don't need to consider the `remote -> console` transition.